gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -2616,7 +2620,58 @@ static GumVirtualizationRequirements
 gum_exec_block_virtualize_sysenter_insn (GumExecBlock * block,
                                          GumGeneratorContext * gc)
 {
+#ifdef HAVE_LINUX
+  return gum_exec_block_virtualize_linux_sysenter (block, gc);
+#else
   return GUM_REQUIRE_RELOCATION;
+#endif
+}
+
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_linux_sysenter (GumExecBlock * block,
+                                          GumGeneratorContext * gc)
+{
+  GumArm64Writer * cw = gc->code_writer;
+  const cs_insn * insn = gc->instruction->ci;
+  gconstpointer continue_normally = cw->code + 1;
+  gconstpointer end_payload = cw->code + 2;
+  const guint32 mrs_x15_nzcv = 0xd53b420f;
+  const guint32 msr_nzcv_x15 = 0xd51b420f;
+
+  gum_arm64_relocator_skip_one (gc->relocator);
+
+  if (gc->opened_prolog != GUM_PROLOG_NONE)
+    gum_exec_block_close_prolog (block, gc);
+
+  gum_arm64_writer_put_stp_reg_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_X17,
+      ARM64_REG_SP, -(16 + GUM_RED_ZONE_SIZE), GUM_INDEX_PRE_ADJUST);
+  gum_arm64_writer_put_instruction (cw, mrs_x15_nzcv);
+
+  gum_arm64_writer_put_sub_reg_reg_imm (cw, ARM64_REG_X17,
+      ARM64_REG_X8, __NR_clone);
+  gum_arm64_writer_put_cbnz_reg_label (cw, ARM64_REG_X17, continue_normally);
+
+  gum_arm64_writer_put_instruction (cw, msr_nzcv_x15);
+  gum_arm64_writer_put_ldp_reg_reg_reg_offset (cw, ARM64_REG_X15,
+      ARM64_REG_X17, ARM64_REG_SP, 16 + GUM_RED_ZONE_SIZE,
+      GUM_INDEX_POST_ADJUST);
+  gum_arm64_writer_put_bytes (cw, insn->bytes, 4);
+  gum_arm64_writer_put_cbnz_reg_label (cw, ARM64_REG_X0, end_payload);
+
+  /* We are on the child return to the original next instruction */
```

Comment:
```suggestion
  /*
   * We are on the child return to the original next instruction.
   *
   * TODO: Is there any way we can avoid clobbering X17 here?
   */
```

Here we're clobbering the child's X17. Can't think of a better way, though, so we should probably leave a comment here.